File: src/zork/models/items/actions/NukeAction.java
package zork.models.items.actions;

import zork.interfaces.IUseAble;
import zork.models.entities.Player;

/**
 * Action for the nuke item. Yep it does what it sounds like it does.
 *
 * @author d.peters
 */
public class NukeAction implements IUseAble {
  /**
   * Use without params. Not supported.
   */
  @Override
  public void use() {
    throw new UnsupportedOperationException();
  }

  /**
   * Only players with special names shall survive the ensuing fallout.
   *
   * @param player the player object
   */
  @Override
  public void use(Player player) {

    switch (player.getName()) {
      case "Chuck Norris":
        System.out.println("\nThat nuclear fallout was just your fart. Nothing to worry about.");
        break;
      case "John Cena":
        System.out.println("\nAnd his name is JOHN CENA, NUKER OF WORLDS!!!");
        break;
      case "Fallout Guy":
        System.out.println("\nAfter years of stasis, you finally wake up in Vault 81.");
        break;
      default:
        System.out.println("\nYou moron! You obliterated everything!");
        player.die();
        break;
    }
  }
}

File: src/main/kotlin/ch/d/peters/zork/models/items/actions/NukeAction.kt
package ch.d.peters.zork.models.items.actions

import ch.d.peters.zork.interfaces.IItemAction
import ch.d.peters.zork.interfaces.IUserInterface

/**
 * Action for the nuke item. Yep it does what it sounds like it does.
 *
 * @author Daniel Peters
 * @version 1.1.0
 */
class NukeAction(private val ui: IUserInterface) : IItemAction {
  /**
   * Only players with special names shall survive the ensuing fallout.
   */
  override fun execute() {
    /*when (player.name) {
       "Chuck Norris" -> ui.displayMessage("\nThat nuclear fallout was just your fart. Nothing to worry about.")
       "John Cena" ->ui.displayMessage("\nAnd his name is JOHN CENA, NUKER OF WORLDS!!!")
       "Fallout Guy" ->
      ui.displayMessage("\nAfter years of stasis, you finally wake up in Vault 81.")
      else -> {
        ui.displayMessage("\nYou moron! You obliterated everything!")
        player.die()
      }
    }*/
  }
}
